File: 1. Easy/The travelling salesman problem/Solution.py
import sys
import math

class town:
    def __init__(self, *townData):
        townData = townData[0].split()
        self.x, self.y = [int(i) for i in townData]
    
n = int(input())
select = town(input())
firstTown = select
result = 0
towns = []

for i in range(n - 1):
    towns.append(town(input()))
    
while len(towns):
    x0, y0 = select.x, select.y
    bestDistance = 10000
    
    for i in range(len(towns)):
        x1, y1 = towns[i].x, towns[i].y
        distance = ((abs(x0 - x1) ** 2 + abs(y0 - y1) ** 2) ** 0.5)
        
        if distance < bestDistance:
            bestDistance = distance
            select = towns[i]
            idx = i
    
    del towns[idx] 
    result += bestDistance

x0, y0 = select.x, select.y    
x1, y1 = firstTown.x, firstTown.y
result += ((abs(x0 - x1) ** 2 + abs(y0 - y1) ** 2) ** 0.5)
    
print(round(result))

File: 1. Easy/The travelling salesman problem/Solution.cpp
#include <cmath>
#include <vector>
#include <iostream>

using namespace std;

struct Position
{
    int x, y;
    
    Position(int x, int y): x(x), y(y) {}
};   

int number;
double result = 0;
vector<Position> towns;

double squaredDistance(const Position& one, const Position& two) {
    return (one.x - two.x) * (one.x - two.x) + (one.y - two.y) * (one.y - two.y);
}

void input() {
    cin >> number; cin.ignore();
    
    for (int i = 0; i < number; i++)
    {
        int x, y;
        cin >> x >> y; cin.ignore();
        towns.emplace_back(Position(x, y));
    }
}

void simpleGreedyAlgorithm() {
    Position firstTown = towns[0];
    
    while (number - 1) {
        int minimalIndex;
        double minimalDistance = 20000.0f;
        
        for (int i = 1; i < number; i++)
        {
            double distance = squaredDistance(towns[0], towns[i]);
                                  
            if (distance < minimalDistance)
            {
                minimalDistance = distance;
                minimalIndex = i;                
            }
        }
        
        swap(towns[0], towns[minimalIndex]);
        towns.erase(towns.begin() + minimalIndex);
        result += sqrt(minimalDistance);
        number--;
    }
    
    result += sqrt(squaredDistance(towns[0], firstTown));
}
    
int main() {
    input();
    simpleGreedyAlgorithm();    
    cout << round(result);
}